server.go

```diff
@@ -1864,6 +1847,10 @@ func (s *Server) GracefulStop() {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
+	if !graceful {
+		s.closeServerTransports()
+	}
```

Comment:
I do believe this should be:

```go
if graceful {
	s.drainAllServerTransports()
} else {
	s.closeServerTransports()
}
```

Reply:
I think I understood it now:
`closeServerTransports` is only called on a graceful stop here because it will abruptly terminate the connections.
In graceful mode, the connections will be closed in `*http2Server.HandleStreams` instead, after `EOF` on the connection